transport/resolver.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"net"
+	"net/url"
+)
+
+// Resolver is a function that resolves a url to a list of IP addresses.
+type Resolver func(url *url.URL) ([]net.Addr, error)
```

Comment:
Let's use an actual interface to be consistent with the Dialers and Conns and our other types.

The function must take a string address, and return a list of string addresses. We may have something like `www.rferl.org` -> `e4887.dscb.akamaiedge.net`, or `www.rferl.org` -> `[104.102.138.83, 2600:141b:13:7ad::1317, 2600:141b:13:7a7::1317]`, depending on the implementation.

We don't really use net.Addr, it's cumbersome, so we stick to strings instead.

Reply:
We can discuss more at the brainstorm, but I’ve been thinking that for our current user cases we may just need a host mapping dialer. We can potentially punt on the resolver interface